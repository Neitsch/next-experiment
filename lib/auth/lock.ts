import { setSecret } from "./index";

import uuid from "uuid";

const getLock = options => {
  const config = require("./config.json");
  const Auth0Lock = require("auth0-lock").default;
  return new Auth0Lock(
    config.AUTH0_CLIENT_ID,
    config.AUTH0_CLIENT_DOMAIN,
    options,
  );
};

const getBaseUrl = () => `${window.location.protocol}//${window.location.host}`;

const getOptions = container => {
  const secret = uuid.v4();
  setSecret(secret);
  return {
    auth: {
      params: {
        scope: "openid",
        state: secret,
      },
      redirectUrl: `${getBaseUrl()}/auth/signed-in`,
      responseType: "token id_token",
    },
    closable: false,
    container,
  };
};

export const show = container => getLock(getOptions(container)).show();
export const logout = () => getLock({}).logout({ returnTo: getBaseUrl() });
