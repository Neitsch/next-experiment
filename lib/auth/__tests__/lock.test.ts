describe("get lock", () => {
  test("works", () => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.doMock("../config.json", () => ({
      AUTH0_CLIENT_DOMAIN: "client_domain",
      AUTH0_CLIENT_ID: "client_id",
    }));
    jest.doMock("../index");
    jest.doMock("uuid");
    const lockConstructor = jest
      .fn()
      .mockImplementation(() => ({ show: jest.fn() }));
    jest.doMock("auth0-lock", () => ({
      default: lockConstructor,
    }));
    require("../lock").show("my-container");
    expect(lockConstructor.mock.calls).toMatchSnapshot();
  });
});

describe("logout", () => {
  test("works", () => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.doMock("../config.json", () => ({
      AUTH0_CLIENT_DOMAIN: "client_domain",
      AUTH0_CLIENT_ID: "client_id",
    }));
    jest.doMock("../index");
    jest.doMock("uuid");
    const lockConstructor = jest
      .fn()
      .mockImplementation(() => ({ logout: jest.fn() }));
    jest.doMock("auth0-lock", () => ({
      default: lockConstructor,
    }));
    require("../lock").logout();
    // expect(() => require("../lock").logout()).not.toThrow();
  });
});
