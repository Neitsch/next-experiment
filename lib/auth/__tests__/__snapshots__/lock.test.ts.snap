// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get lock works 1`] = `
Array [
  Array [
    "client_id",
    "client_domain",
    Object {
      "auth": Object {
        "params": Object {
          "scope": "openid",
          "state": undefined,
        },
        "redirectUrl": "about:///auth/signed-in",
        "responseType": "token id_token",
      },
      "closable": false,
      "container": "my-container",
    },
  ],
]
`;
